sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DatasourceCredentialsTests.cs

```diff
@@ -56,6 +56,28 @@ public async Task DatasourceCredentialSendsSecretDuringCreation(DatasourceCreden
             Assert.That(content, Contains.Substring(expectedSubstring));
         }
 
+        [Test]
+        public async Task ServicePrincipalDatasourceCredentialSendsSecretDuringUpdate()
+        {
+            MockResponse updateResponse = new MockResponse(200);
+            updateResponse.SetContent(DatasourceCredentialResponseContent);
+
+            MockTransport mockTransport = new MockTransport(updateResponse);
+            MetricsAdvisorAdministrationClient adminClient = CreateInstrumentedAdministrationClient(mockTransport);
+
+            var credential = new ServicePrincipalDatasourceCredential(DataSourceCredentialType.ServicePrincipal, FakeGuid,
+                default, default, new ServicePrincipalParam("mock", "mock"));
+
+            credential.UpdateClientSecret("secret");
```

Comment:
would it be good to validate that before this `secret=default`?

Reply:
Updated.